core/src/services/lakefs/backend.rs

```diff
@@ -0,0 +1,324 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
+use bytes::Buf;
+use chrono::{TimeZone, Utc};
+use http::Response;
+use http::StatusCode;
+use log::debug;
+use serde::Deserialize;
+use serde::Serialize;
+
+use super::core::LakefsCore;
+use super::core::LakefsStatus;
+use super::error::parse_error;
+use crate::raw::*;
+use crate::*;
+
+/// Configuration for Lakefs service support.
+#[derive(Default, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct LakefsConfig {
```

Comment:
Please place the lake config in a separate file. It appears you are referring to old code. Perhaps you could rebase with the latest main branch.